test/compiler/inline.jl

```diff
@@ -1709,3 +1709,10 @@ let src = code_typed1((Any,)) do x
     end
     @test count(iscall((src, f_union_unmatched)), src.code) == 0
 end
+
+# Test that inlining doesn't unnecesarily move things to stmt position
+@noinline f_no_inline_invoke(x::Union{Symbol, Nothing}=nothing) = Base.donotdelete(x)
+g_no_inline_invoke(x) = f_no_inline_invoke(x)
+let src = code_typed1(g_no_inline_invoke, Tuple{Any})
+    @test count(x->isa(x, GlobalRef), src.code) == 0
+end
```

Comment:
I don't think this test case is actually asserting the improvement from this PR. It produces the same IR on master and on this PR:
```julia
julia> @noinline f_no_inline_invoke(x::Union{Symbol, Nothing}=nothing) = Base.donotdelete(x)
f_no_inline_invoke (generic function with 2 methods)

julia> g_no_inline_invoke(x) = f_no_inline_invoke(x)
g_no_inline_invoke (generic function with 1 method)

julia> code_typed(g_no_inline_invoke, (Any,))
1-element Vector{Any}:
 CodeInfo(
1 ─ %1 = Main.f_no_inline_invoke(x)::Nothing
└──      return %1
) => Nothing
```